great_expectations/datasource/fluent/spark_file_path_datasource.py

```diff
@@ -36,6 +36,25 @@ def _get_reader_options_include(self) -> set[str] | None:
         return {"header", "infer_schema"}
 
 
+class DirectoryCSVAsset(_FilePathDataAsset):
+    # Overridden inherited instance fields
+    type: Literal["directory_csv"] = "directory_csv"
+    data_directory: str
```

Comment:
Can we make this a `pathlib.Path` or better yet a `DirectoryPath` pydantic type.
This will ensure the value exists and is a directory, and error if it isn't.

https://docs.pydantic.dev/latest/usage/types/#pydantic-types

Reply:
Just to follow up, went with pathlib.Path since the data_directory is relative to the base_path. In a future PR we can validate the full path (base_path + data_directory)